src/components/DataTable/DataTable.tsx

```diff
@@ -114,94 +105,35 @@ class DataTable extends React.PureComponent<CombinedProps, DataTableState> {
   }
 
   render() {
+    const {headings, totals, rows, footerContent} = this.props;
     const {
-      columnContentTypes,
-      headings,
-      totals,
-      rows,
-      truncate,
-      footerContent,
-      sortable,
-      defaultSortDirection = 'ascending',
-      initialSortColumnIndex = 0,
-    } = this.props;
-
-    const {
-      collapsed,
+      condensed,
       columnVisibilityData,
-      heights,
-      sortedColumnIndex = initialSortColumnIndex,
-      sortDirection = defaultSortDirection,
       isScrolledFarthestLeft,
       isScrolledFarthestRight,
     } = this.state;
 
     const className = classNames(
       styles.DataTable,
-      collapsed && styles.collapsed,
-      footerContent && styles.hasFooter,
+      condensed && styles.condensed,
     );
 
     const wrapperClassName = classNames(
       styles.TableWrapper,
-      collapsed && styles.collapsed,
+      condensed && styles.condensed,
     );
 
-    const footerClassName = classNames(footerContent && styles.TableFoot);
-
-    const footerMarkup = footerContent ? (
-      <tfoot className={footerClassName}>
-        <tr>{this.renderFooter()}</tr>
-      </tfoot>
-    ) : null;
+    const headingMarkup = <tr>{headings.map(this.renderHeadings)}</tr>;
 
     const totalsMarkup = totals ? (
       <tr>{totals.map(this.renderTotals)}</tr>
     ) : null;
 
-    const headingMarkup = (
-      <tr>
-        {headings.map((heading, headingIndex) => {
```

Comment:
(extracted this anonymous callback into a class method like the other render{Row} callbacks, see 126 and 246)